lib/iris/tests/__init__.py

```diff
@@ -695,18 +703,50 @@ def check_graphic(self, tol=_DEFAULT_IMAGE_TOLERANCE):
 
             figure.savefig(result_fname)
 
-            if not os.path.exists(expected_fname):
-                warnings.warn('Created image for test %s' % unique_id)
+            # XXX: Deal with a new test result i.e. it's not in the repo
+
+            # hash the created image using sha1
+            with open(result_fname, 'rb') as res_file:
+                sha1 = hashlib.sha1(res_file.read())
+
+            emsg = None
+
+            try:
+                uris = repo[unique_id]
+            except KeyError:
+                # This test is not registered in the image json store,
+                # therefore treat it as a *new* test to be registered.
+                dname = os.path.join(os.path.dirname(__file__),
+                                     'results', 'visual_tests')
+                expected_fname = os.path.join(dname, unique_id+'.png')
+                if not os.path.isdir(dname):
+                    os.makedirs(dname)
+                wmsg = 'Created image for test {}'
+                warnings.warn(wmsg.format(unique_id))
                 shutil.copy2(result_fname, expected_fname)
-
-            err = mcompare.compare_images(expected_fname, result_fname, tol=tol)
+            else:
+                # Cherry-pick the registered expected hashes from the
+                # test case uri/s.
+                expected = []
```

Comment:
I think a list comprehension would be clearer than init + append here.
Seems an obvious edit of what is here already.